### ISSUE TITLE
Use same networks for 1) comparison with classical approach and 2) feature network

### ISSUE THREAD

**Issue by psaegert** [MrWhatZitToYaa/IGNNS-final-project#24]
@CKleiber and @psaegert discussed our plans of comparing the CNF + feature network against their classical version, and use the same network for both tasks for best comparability.

We thought about optimizing the feature networks with direct squared loss training $y = F(X)$ since it would be faster than optimizing it while attached to a CNF.
@psaegert thinks that optimizing the feature networks and CNF jointly might not be necessary since the CNF hyperparameters mainly depend on the complexity of the simulation parameters, which do not change when the feature network / condition changes.
We then use the same (optimized) model hyperparameters for the classical training and the feature network for CNF training.

Hyperparameter optimization would involve these networks:

**1. Trajectory $\rightarrow$ Parameters Setting**
- FCNN: y = FCNN(X), standard approach
- RNN: y = RNN(X), motivated by the time-series structure of the data

**2. Video $\rightarrow$ Parameters Setting**
- ConvNN + FCNN, additional convolutional neural network to detect features from the frames of the video
- ConvNN + RNN

The comparison scope would be as follows:

### 1. Trajectory $\rightarrow$ Parameters  Setting
**FCNN**

_Hypothesis: Better predictions with CNF. Predicting posteriors instead of point estimates is a plus._

classical FCNN **vs** CNF (FCNN as feature network)

**RNN**
_Hypothesis: RNNs should work better than FCNN since they are designed for time-series data_

classical RNN **vs** CNF (RNN as feature network)

### 2. Video $\rightarrow$ Parameters  Setting
_Hypothesis: Worse performance due to additional task of identifying the ball from the video_

**ConvNN + FCNN**
classical ConvNN + FCNN **vs** CNF (ConvNN + FCNN as feature network)

**ConvNN + RNN**
classical ConvNN + RNN **vs** CNF (ConvNN + RNN as feature network)
